File: CS/WebSite/DevExpress/ASPxSpellCheckerForms/SpellCheckForm.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;
using DevExpress.Web.ASPxSpellChecker;
using DevExpress.Web.ASPxEditors;
using DevExpress.Web.ASPxSpellChecker.Localization;
using DevExpress.Web.ASPxClasses.Internal;

public partial class SpellCheckForm : SpellCheckerUserControl {
    protected override void PrepareChildControls() {
        PrepareCheckedDiv();
        PrepareChangeToPanel();
        PrepareAddToDictionaryButton();
        Localize();

        base.PrepareChildControls();
    }
    protected override ASPxEditBase[] GetChildDxEdits() {
        return new ASPxEditBase[] { txtChangeTo, SCSuggestionsListBox };
    }

    protected override ASPxButton[] GetChildDxButtons() {
        return new ASPxButton[] {
            btnAddToDictionary,
            btnChange,
            btnChangeAll,
            btnClose,
            btnIgnore,
            btnIgnoreAll,
            btnOptions
        };
    }

    protected void PrepareAddToDictionaryButton() {
        btnAddToDictionary.Enabled = SpellChecker.GetCustomDictionary() != null;
    }
    protected void PrepareCheckedDiv() {
        WebControl checkedDiv = RenderUtils.CreateDiv();
        checkedDiv.ID = "SCCheckedDiv";
        SCCheckedDivPlaceHolder.Controls.Add(checkedDiv);

        CheckedTextContainerStyle style = SpellChecker.GetCheckedTextContainerStyle();
        checkedDiv.Attributes.Add("class", style.CssClass);
        CssStyleCollection styles = style.GetStyleAttributes(Page);
        foreach (string key in styles.Keys)
            checkedDiv.Style.Add(key, styles[key]);
    }
    protected void PrepareChangeToPanel() {
        ClientIDHelper.EnableClientIDGeneration(txtChangeTo);
        ClientIDHelper.EnableClientIDGeneration(SCSuggestionsListBox);
    }
	protected void Localize() {
		btnIgnore.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreOnceButton);
		btnIgnoreAll.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreAllButton);
		btnAddToDictionary.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.AddToDictionaryButton);
		btnChange.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.ChangeButton);
		btnChangeAll.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.ChangeAllButton);
		btnOptions.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.OptionsButton);
		btnClose.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.CloseButton);
	}
}

File: VB/WebSite/DevExpress/ASPxSpellCheckerForms/SpellCheckForm.ascx.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Data
Imports System.Configuration
Imports System.Collections
Imports System.Web
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls
Imports DevExpress.Web.ASPxSpellChecker
Imports DevExpress.Web.ASPxEditors
Imports DevExpress.Web.ASPxSpellChecker.Localization
Imports DevExpress.Web.ASPxClasses.Internal

Partial Public Class SpellCheckForm
	Inherits SpellCheckerUserControl
	Protected Overrides Sub PrepareChildControls()
		PrepareCheckedDiv()
		PrepareChangeToPanel()
		PrepareAddToDictionaryButton()
		Localize()

		MyBase.PrepareChildControls()
	End Sub
	Protected Overrides Function GetChildDxEdits() As ASPxEditBase()
		Return New ASPxEditBase() { txtChangeTo, SCSuggestionsListBox }
	End Function

	Protected Overrides Function GetChildDxButtons() As ASPxButton()
		Return New ASPxButton() { btnAddToDictionary, btnChange, btnChangeAll, btnClose, btnIgnore, btnIgnoreAll, btnOptions }
	End Function

	Protected Sub PrepareAddToDictionaryButton()
		btnAddToDictionary.Enabled = SpellChecker.GetCustomDictionary() IsNot Nothing
	End Sub
	Protected Sub PrepareCheckedDiv()
		Dim checkedDiv As WebControl = RenderUtils.CreateDiv()
		checkedDiv.ID = "SCCheckedDiv"
		SCCheckedDivPlaceHolder.Controls.Add(checkedDiv)

		Dim style As CheckedTextContainerStyle = SpellChecker.GetCheckedTextContainerStyle()
		checkedDiv.Attributes.Add("class", style.CssClass)
		Dim styles As CssStyleCollection = style.GetStyleAttributes(Page)
		For Each key As String In styles.Keys
			checkedDiv.Style.Add(key, styles(key))
		Next key
	End Sub
	Protected Sub PrepareChangeToPanel()
		ClientIDHelper.EnableClientIDGeneration(txtChangeTo)
		ClientIDHelper.EnableClientIDGeneration(SCSuggestionsListBox)
	End Sub
	Protected Sub Localize()
		btnIgnore.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreOnceButton)
		btnIgnoreAll.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.IgnoreAllButton)
		btnAddToDictionary.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.AddToDictionaryButton)
		btnChange.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.ChangeButton)
		btnChangeAll.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.ChangeAllButton)
		btnOptions.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.OptionsButton)
		btnClose.Text = ASPxSpellCheckerLocalizer.GetString(ASPxSpellCheckerStringId.CloseButton)
	End Sub
End Class